preprocessing/formparser.py

```diff
@@ -0,0 +1,847 @@
+import numpy as np
+from copy import copy
+
+
+from preprocessing import constants
+from preprocessing.constants import prefix, suffix
+from preprocessing.constants import months_defaults, working_day
+from preprocessing import utils
+
+
+class FormParser:
+
+    def __init__(self, form=None, verbose=False) -> None:
+        """
+        Initialize the FormParser object.
+
+        :param form: (optional) The form data to be parsed.
+        :type form: dict or None
+
+        Initializes the FormParser object with the given form data. If form is not None,
+        the init_parser method is called to initialize the parser.
+
+        Attributes:
+        - form (dict or None): The form data to be parsed.
+        - formtype (None): The type of the form.
```

Comment:
please indicate which values can be provided here, or link to the variable in preprocessing/constants.py which holds this information